hedera-node/src/main/java/com/hedera/services/ServicesState.java

```diff
@@ -178,15 +183,21 @@ public void init(Platform platform, AddressBook addressBook) {
 					new FCMap<>(new MerkleEntityId.Provider(), MerkleToken.LEGACY_PROVIDER));
 			setChild(ChildIndices.TOKEN_ASSOCIATIONS,
 					new FCMap<>(MerkleEntityAssociation.LEGACY_PROVIDER, MerkleTokenRelStatus.LEGACY_PROVIDER));
+			setChild(ChildIndices.SPECIAL_FILE_SYSTEM, new SpecialFileSystem(ctx.nodeAccount()));
 			log.info("Init called on Services node {} WITHOUT Merkle saved state", nodeId);
 		} else {
 			log.info("Init called on Services node {} WITH Merkle saved state", nodeId);
 			merkleDigest.accept(this);
+			// if loaded from state, SpecialFileSystem is created with non-args constructor
+			// re-init SpecialFileSystem so it has nodeAccountID setup correctly
+			SpecialFileSystem preSpecialFileSystem = getChild(ChildIndices.SPECIAL_FILE_SYSTEM);
+			setChild(ChildIndices.SPECIAL_FILE_SYSTEM,
+					new SpecialFileSystem(preSpecialFileSystem.getFileMap(),
+							EntityIdUtils.asLiteralString(ctx.nodeAccount())));
 			printHashes();
 		}
 
-		var properties = new StandardizedPropertySources(bootstrapProps, loc -> new File(loc).exists());
-		ctx = new ServicesContext(nodeId, platform, this, properties);
+
```

Comment:
I moved ctx initialization before 
`if (getNumberOfChildren() < ChildIndices.SPECIAL_FILE_SYSTEM)`
since SpecialFileSystem constructor need to use node id as parameter